src/drivers/magnetometer/isentek/ist8310/IST8310.cpp

```diff
@@ -85,6 +85,15 @@ void IST8310::print_status()
 
 int IST8310::probe()
 {
+	// Apparently, the IST8310's WHOAMI register is writeable. Presumably,
+	// this can get corrupted by bus noise. It is only reset if powered off
+	// for 30s or by a reset.
+	//
+	// Therefore, we do a reset before doing the WHOAMI.
+	RegisterWrite(Register::CNTL2, CNTL2_BIT::SRST);
+
+	px4_usleep(10000);
```

Comment:
Where are these times coming from? 

What about the 50 ms for reset from the datasheet? https://github.com/PX4/PX4-Autopilot/blob/1175d65754193f9c4b2ea52fc7ea7149232d2156/src/drivers/magnetometer/isentek/ist8310/IST8310.cpp#L119

Reply:
You could also try multiple times in a loop (up to a max of 50ms) to try and keep it short, and only do it if the WAI read fails in the first place.

Reply:
The value comes from the ArduPilot PR:
https://github.com/ArduPilot/ardupilot/pull/26929/files#diff-556a4995584d1fd43c0b5cb6cf5a8bd04adbb86acc94c2ac94c2749264dc12d4R130

Reply:
New IST8310 version datasheet ![image](https://github.com/PX4/PX4-Autopilot/assets/46874772/61d46e35-e2da-4d9a-8afe-5e3a860bb153)

Reply:
I can either try to reproduce this and try to optimize this sleep value, or I can trust what ArduPilot has ended up with and merge it.

Reply:
We tried to reproduce this issue with our own hardware with PX4, but was unable to. We were able to reproduce and confirm the fix using Ardupilot and https://github.com/ArduPilot/ardupilot/pull/26929. 

This issue can only be reproduce with certain FC + Compass combination, and it appear that it can not be reproduce with our FC. 

There is a user with their own custom FC that can reproduce this IST8310 issue, but it is difficult to communicate and test these type of non-stable FW.

Reply:
I would agree and say 50ms to be safe since that is what the datasheet specifies. If you're too early you end up with no baro. An extra 40ms at boot won't be noticed

Reply:
It will add up. It's often probed on multiple buses.